esmvalcore/_data_finder.py

```diff
@@ -95,16 +103,29 @@ def get_start_end_year(filename):
                 time = cube.coord('time')
             except iris.exceptions.CoordinateNotFoundError:
                 continue
-            start_year = time.cell(0).point.year
-            end_year = time.cell(-1).point.year
+            start_point = time.cell(0).point
+            start_datetime = datetime.datetime(
+                start_point.year, start_point.month, start_point.day,
+                start_point.hour, start_point.minute, start_point.second)
+            start_date = isodate.date_isoformat(
+                start_datetime, format=isodate.isostrf.DATE_BAS_COMPLETE)
+            start_year = start_point.year
+
+            end_point = time.cell(-1).point
+            end_datetime = datetime.datetime(
+                end_point.year, end_point.month, end_point.day,
+                end_point.hour, end_point.minute, end_point.second)
+            end_date = isodate.date_isoformat(
+                end_datetime, format=isodate.isostrf.DATE_BAS_COMPLETE)
+            end_year = end_point.year
             break
 
     if start_year is None or end_year is None:
         raise ValueError(f'File {filename} dates do not match a recognized'
                          'pattern and time can not be read from the file')
 
     logger.debug("Found start_year %s and end_year %s", start_year, end_year)
-    return int(start_year), int(end_year)
+    return int(start_year), int(end_year), start_date, end_date
```

Comment:
Why not just return the full date and then just access the year outside?

Reply:
Sure, I'll change that

Reply:
Actually returning both comes in quite handy

Reply:
Yes, but it is really ugly

Reply:
start_date and end_date are strings, accessing the year outside after every time this function is called in not that direct.

Reply:
We are keeping it as a string to avoid having to parse the dates